### PR TITLE
Merge channel flow demo cfn template to the main chat app demo cfn te…

### PR DESCRIPTION
**Pull request by eliasdem**
…mplate

**Issue #:**

**Description of changes:**

**Testing**

1. How did you test these changes? 
  Deployed the updated cfn template `template.yaml` and verified the channel flows features
2. Can these changes be tested using one of the demo application? If yes, which demo application can be used to test it?
  Yes, the chat demo app.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apps/chat/src/backend/serverless/template.yaml [hunk 1]
```diff
@@ -190,6 +302,28 @@ Resources:
               - "x-amz-id-2"
             MaxAge: "3000"
 
+  # Creates a role that allows Cognito to send SNS messages
```
Comment (sujayv-amzn) [inline apps/chat/src/backend/serverless/template.yaml:305]: This role is not used anywhere. Can be removed

File: apps/chat/src/backend/serverless/template.yaml [hunk 2]
```diff
@@ -35,138 +35,250 @@ Resources:
...
+      Layers:
+        - !Ref AWSSDKChimeLayer
+      Code:
+        ZipFile: >
+          "use strict";
+          const AWS = require("aws-sdk");
+          const uuidv4 = require("uuid");
+          const response = require("cfn-response");
+          AWS.config.update({ region: process.env.AWS_REGION });
+          const chime = new AWS.Chime({ region: process.env.AWS_REGION });
```
Comment (sujayv-amzn) [inline apps/chat/src/backend/serverless/template.yaml:227]: Will you be updating this to use the new Identity and Messaging SDKs?

File: apps/chat/src/backend/serverless/template.yaml [hunk 3]
```diff
@@ -35,138 +35,250 @@ Resources:
...
+    DependsOn:
+      - "ProfanityAndDLPProcessor"
+    Properties:
+      FunctionName: !Sub ${DemoName}-ProfanityAndDLPProcessor
+      Action: "lambda:InvokeFunction"
+      Principal: "chime.amazonaws.com"
+      SourceAccount: {Ref: "AWS::AccountId"}
+      SourceArn: !Sub "arn:aws:chime:${AWS::Region}:${AWS::AccountId}:app-instance/*"
+
+  #Lambda that creates AWS Chime App Instance and Channel Flow resources as one off on stack creation
```
Comment (LeviSklaroff) [inline apps/chat/src/backend/serverless/template.yaml:113]: nit. Should have a space before `Lambda`

File: apps/chat/src/backend/serverless/template.yaml [hunk 4]
```diff
@@ -35,138 +35,250 @@ Resources:
...
   TriggerChimeAppInstanceLambda:
-      Type: AWS::CloudFormation::CustomResource
-      Properties: 
-        ServiceToken: !GetAtt ChimeAppInstanceLambda.Arn
+    Type: AWS::CloudFormation::CustomResource
+    DependsOn: ProfanityAndDLPProcessor
+    Properties:
+      ServiceToken: !GetAtt ChimeAppInstanceLambda.Arn
 
   #Lambda that creates AWS Chime App admin user
```
Comment (LeviSklaroff) [inline apps/chat/src/backend/serverless/template.yaml:209]: nit. spacing

File: apps/chat/src/backend/serverless/template.yaml [hunk 5]
```diff
@@ -35,138 +35,250 @@ Resources:
...
+              } catch (error) {
+                console.log("ERROR CAUGHT \n", error);
                 await response.send(event, context, response.FAILED, {});
               }
-            };
-        Environment:
-          Variables:
-            CHIME_APP_INSTANCE_ARN: !GetAtt TriggerChimeAppInstanceLambda.AppInstanceArn 
+            } else {
+              //NOT A CREATE REQUEST
```
Comment (LeviSklaroff) [inline apps/chat/src/backend/serverless/template.yaml:269]: nit. spacing

File: apps/chat/src/backend/serverless/template.yaml (hunk 1 above)
Comment (eliasdem) [reply apps/chat/src/backend/serverless/template.yaml]: updated

File: apps/chat/src/backend/serverless/template.yaml (hunk 2 above)
Comment (eliasdem) [reply apps/chat/src/backend/serverless/template.yaml]: updated

**Review by LeviSklaroff — DISMISSED** [review]

**Review by LeviSklaroff — APPROVED** [review]

**Review by sujayv-amzn — APPROVED** [review]

**Pull request merged**